tests/test_pipeline.py

```diff
@@ -0,0 +1,121 @@
+"""Fondant pipelines test."""
+from pathlib import Path
+
+import pytest
+
+from fondant.exceptions import InvalidPipelineDefinition
+from fondant.pipeline import FondantComponentOp, FondantPipeline
+
+valid_pipeline_path = Path(__file__).parent / "example_pipelines/valid_pipeline"
+invalid_pipeline_path = Path(__file__).parent / "example_pipelines/invalid_pipeline"
+
+
+@pytest.fixture
+def default_pipeline_args():
+    return {
+        "pipeline_name": "pipeline",
+        "base_path": "gcs://bucket/blob",
+    }
+
+
+@pytest.mark.parametrize(
+    "valid_pipeline_example",
+    [
+        (
+            "example_1",
+            ["first_component.yaml", "second_component.yaml", "third_component.yaml"],
+        ),
+    ],
+)
+def test_valid_pipeline(default_pipeline_args, valid_pipeline_example, tmp_path):
```

Comment:
Your test here doesn't catch the issue that you're validating against the output manifest because each input subset is always propagated as an output subset. You might want to create an example where this is not the case.

Reply:
I'v adjusted the examples, now `captions` are outputted in the first component and are not propagated in the second one. I later invoke it in the third one

Reply:
They are still propagated, and you will not catch the issue with this test. I'll add a commit with a test that fails to show the issue.